rpc/src/v1/impls/eth.rs

```diff
@@ -639,8 +639,25 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		Box::new(future::done(self.rich_block(BlockId::Hash(hash.into()).into(), include_txs)))
 	}
 
+	fn check_block_existence(&self, num: BlockNumber) -> Result<()> {
+		if let BlockNumber::Num(number) = num {
+			match self.client.chain_info().best_block_number {
+				BlockNumber::Num(best_block_number) => {
+					if num < best_block_number {
```

Comment:
This is quite wrong, actually this are the only queries that make sense at all, since:
1. In database we store blocks from `[0...best_block_number]`
2. Querying anything from that range should succeed
3. Querying anything greater than `best_block_number` will always fail

With ancient blocks the case is that the database is not complete, so we have a situation where:
1. We have `[0..best_ancient_block] ... [first_block .. best_block]`
2. The latter range has been restored together with the snapshot
3. The former range has been downloaded during ancient block import
4. Any query that tries to get `num < best_block_number` and fails (i.e. returns `None`) most likely falls into `[best_ancient_block .. first_block]` range, so we should detect that and return the error only in such case.